### PR TITLE
Refactor bookmarksToolbar.js with Aphrodite

### LINKED ISSUE #7920: Refactor bookmarksToolbar.js with Aphrodite
## Test Plan
1. Automated test should be passing (CXX=g++-4.8 NODE_ENV=test TEST_DIR=bookmark-components)
2. Bookmark several pages if you have not
3. Test that creating a bookmark on the bookmarks toolbar works
4. Test that creating a bookmark folder on the bookmarks toolbar works
5. Test that moving a bookmark into a folder by drag and drop on the bookmarks folder works
6. Test that clicking a bookmark in the toolbar loads the bookmark.
7. Test that clicking a bookmark in a bookmark toolbar folder loads the bookmark.
8. Change the window size
9. Make sure the overflow indicator is displayed on the right of the bookmarks toolbar

----

**Describe the issue you encountered:** Refactor bookmarksToolbar.js with Aphrodite.

1st attempt: https://github.com/luixxiul/browser-laptop/commit/abbd11c3715192617e981493943483a3de28e95f

- Extra QA steps:
    1.
    2.
    3.

- Screenshot if needed:

- Any related issues:

### PR DESCRIPTION
**Pull request by luixxiul**
## Test Plan
1. Automated test should be passing (CXX=g++-4.8 NODE_ENV=test TEST_DIR=bookmark-components)
2. Bookmark several pages if you have not
3. Test that creating a bookmark on the bookmarks toolbar works
4. Test that creating a bookmark folder on the bookmarks toolbar works
5. Test that moving a bookmark into a folder by drag and drop on the bookmarks folder works
6. Test that clicking a bookmark in the toolbar loads the bookmark.
7. Test that clicking a bookmark in a bookmark toolbar folder loads the bookmark.
8. Change the window size
9. Make sure the overflow indicator is displayed on the right of the bookmarks toolbar

## Description
Closes #7920

bookmarksToolbarTest.js is updated thanks to the help of @NejcZdovc: https://github.com/luixxiul/browser-laptop/commit/abbd11c3715192617e981493943483a3de28e95f#commitcomment-21504404

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).

### CODE REVIEW

File: app/renderer/components/bookmarksToolbar.js [hunk 1]
```diff
@@ -200,19 +203,36 @@ class BookmarkToolbarButton extends ImmutableComponent {
       onContextMenu={this.onContextMenu}>
       {
         this.isFolder && this.props.showFavicon
-        ? <span className='bookmarkFavicon bookmarkFolder fa fa-folder-o' style={iconStyle} />
+        ? <span className={css(
+            styles.bookmarkToolbarButton__bookmarkFavicon,
+            styles.bookmarkToolbarButton__bookmarkFolder
+          ) + ' fa fa-folder-o'}
```
Comment (NejcZdovc) [inline app/renderer/components/bookmarksToolbar.js]: Use cx here

File: app/renderer/components/bookmarksToolbar.js [hunk 2]
```diff
@@ -221,7 +241,7 @@ class BookmarkToolbarButton extends ImmutableComponent {
       </span>
       {
         this.isFolder
-        ? <span className='bookmarkFolderChevron fa fa-chevron-down' />
+        ? <span className={css(styles.bookmarkToolbarButton__bookmarkFolderChevron) + ' fa fa-chevron-down'} data-test-id='bookmarkFolderChevron' />
```
Comment (NejcZdovc) [inline app/renderer/components/bookmarksToolbar.js]: Use cx here

File: app/renderer/components/bookmarksToolbar.js [hunk 3]
```diff
@@ -389,19 +409,14 @@ class BookmarksToolbar extends ImmutableComponent {
...
-          allowDragging: this.props.shouldAllowWindowDrag,
-          showFavicon,
-          showOnlyFavicon
-        })
-      }
+      className={css(
+        styles.bookmarksToolbar,
+        this.props.shouldAllowWindowDrag && styles.bookmarksToolbar__allowDragging,
+        styles.bookmarksToolbar__showOnlyFavicon
+      ) + ' showFavicon showOnlyFavicon'}
```
Comment (NejcZdovc) [inline app/renderer/components/bookmarksToolbar.js]: Use cx here

File: app/renderer/components/bookmarksToolbar.js [hunk 4]
```diff
@@ -424,13 +439,120 @@ class BookmarksToolbar extends ImmutableComponent {
...
       {
         this.overflowBookmarkItems.size !== 0
-        ? <Button iconClass='overflowIndicator fa-angle-double-right'
+        ? <Button iconClass='fa-angle-double-right'
           onClick={this.onMoreBookmarksMenu}
-          className='bookmarkButton' />
+          className={css(
+            styles.bookmarksToolbar__bookmarkButton,
+            styles.bookmarksToolbar__overflowIndicator
+          ) + ' bookmarkButton'} />
```
Comment (NejcZdovc) [inline app/renderer/components/bookmarksToolbar.js]: Use cx here

File: app/renderer/components/bookmarksToolbar.js [hunk 5]
```diff
@@ -424,13 +439,120 @@ class BookmarksToolbar extends ImmutableComponent {
...
+    maxWidth: bookmarkItemMaxWidth,
+    padding: `2px ${bookmarkItemPadding}`,
+    textOverflow: 'ellipsis',
+    userSelect: 'none',
+    whiteSpace: 'nowrap',
+    WebkitAppRegion: 'no-drag',
+    WebkitUserSelect: 'none',
+
+    ':hover': {
+      background: 'white',
```
Comment (NejcZdovc) [inline app/renderer/components/bookmarksToolbar.js]: please use HEX code for the color

**Comment by luixxiul** [comment]
@NejcZdovc I updated.

File: app/renderer/components/bookmarksToolbar.js [hunk 6]
```diff
@@ -200,19 +204,43 @@ class BookmarkToolbarButton extends ImmutableComponent {
...
       {
         this.isFolder && this.props.showFavicon
-        ? <span className='bookmarkFavicon bookmarkFolder fa fa-folder-o' style={iconStyle} />
+        ? <span className={css(
+            styles.bookmarkToolbarButton__bookmarkFavicon,
+            styles.bookmarkToolbarButton__bookmarkFolder
+          ) + ' ' + cx({
+            fa: true,
+            'fa-folder-o': true
+          })}
```
Comment (NejcZdovc) [inline app/renderer/components/bookmarksToolbar.js]: I was thinking something like:
```
<span className={cx({
  fa: true,
  'fa-folder-o': true,
  [css(styles.bookmarkToolbarButton__bookmarkFavicon)]: true,
  [css(styles.bookmarkToolbarButton__bookmarkFolder)]: true
})}
```
This way you don't need `+ '  ' +` and it's more readable in my opinion

Comment (luixxiul) [inline app/renderer/components/bookmarksToolbar.js]: I like that. I will update later.

**Pull request merged**